controllers/vspheremachine_controller.go

```diff
@@ -187,28 +189,49 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return reconcile.Result{}, err
 	}
 
-	// Fetch the CAPI Machine and CAPI Cluster.
+	// Fetch the CAPI Machine.
 	machine, err := clusterutilv1.GetOwnerMachine(ctx, r.Client, machineContext.GetObjectMeta())
 	if err != nil {
-		return reconcile.Result{}, err
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get Machine for VSphereMachine")
 	}
 	if machine == nil {
-		log.Info("Waiting on Machine controller to set OwnerRef on VSphereMachine")
+		log.Info("Waiting for Machine controller to set OwnerRef on VSphereMachine")
 		return reconcile.Result{}, nil
 	}
-
 	log = log.WithValues("Machine", klog.KObj(machine))
 	ctx = ctrl.LoggerInto(ctx, log)
-	cluster := r.fetchCAPICluster(ctx, machine, machineContext.GetVSphereMachine())
+
+	// AddOwners adds the owners of Machine as k/v pairs to the logger.
+	// Specifically, it will add KubeadmControlPlane, MachineSet and MachineDeployment.
+	ctx, log, err = clog.AddOwners(ctx, r.Client, machine)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	cluster, err := clusterutilv1.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
+	if err != nil {
+		log.Error(err, "Failed to get Cluster from VSphereCluster: Machine is missing cluster label or cluster does not exist")
+		// Cluster is set to nil to continue for the delete case and return in the regular case.
+		cluster = nil
+	}
+	if cluster != nil {
```

Comment:
What about
```suggestion
	if err != nil || cluster == nil {
		log.Error(err, "Failed to get Cluster from VSphereCluster: Machine is missing cluster label or cluster does not exist")
	}
```

Reply:
Based on previous logic it is not an error if cluster is nil for the VSphereMachine deletion case. So I think we shouldn't log in this non-error case